home_audit.node.js

```diff
@@ -935,8 +935,9 @@ let validationRules = {
         return this._heating_efficiency_method(value, 2);
     },
     _heating_efficiency_method: function(value, num) {
-        if(value && (_homeValues['heating_type_'+num] === 'none' || TypeRules._is_empty(_homeValues['heating_type_'+num]))) {
-            return new Validation('Efficiency method should not be set if heating type is "none" or empty', ERROR);
+        const noEfficiencyMethod = ['baseboard', 'wood_stove', 'none'];
+        if(value && noEfficiencyMethod.indexOf(_homeValues['heating_type_'+num]) > -1 || (_homeValues['heating_type_'+num] === 'central_furnace' && _homeValues['heating_fuel_'+num] === 'electric') || TypeRules._is_empty(_homeValues['heating_type_'+num])) {
+            return new Validation('Efficiency method should not be set if heating type is "central furnace" and fuel is "electric", or if heating type is "baseboard", "wood stove", "none", or empty', ERROR);
```

Comment:
Sorry to leave yet another comment on such a short routine :). But I'm pretty sure we don't allow an efficiency value for any heater than has "electric" as its fuel type, because the efficiency of resistive heating is always the same as a consequence of the physics of electrical resistance. Could we simplify this rule and the text by formulating it as "...should not be set for electric heaters, wood stoves, or systems without heating."?

Reply:
This is validating whether efficiency *method*, which determines whether to enter a value or year, is a valid entry.  Regardless, I am not sure you are right... http://hes-documentation.labworks.org/home/api-definitions/api-methods/submit-inputs/building/systems/hvac/heating

Reply:
Sorry - I forgot that "electric" is the fuel type for HVACs that run on electricity even if they're not resistive heating.